Document/0x05b-Basic-Security_Testing.md

```diff
@@ -1244,13 +1244,37 @@ For enterprise tools, see the section "Static Source Code Analysis" in the chapt
 
 #### Dynamic Analysis
 
--- ToDo: <https://github.com/OWASP/owasp-mstg/issues/1240>
+Dynamic Analysis tests the mobile app by executing and running the app binary and analysing the workflows's for vulnerabilities. For example vulnerabilities regarding data storage might be sometimes hard to catch during static analysis, but in dynamic analysis you can easily spot what information is stored persistently and if the information is protected properly. Besides this, dynamic analysis allows the tester to properly identify:
+
+- Business logic flaws
+- Vulnerabilities in the tested environments
+- Weak input validation and bad input/output encoding as they are processed through one or multiple services
+
+Analysis can be assisted by automated tools, such as [MobSF](https://github.com/MobSF/Mobile-Security-Framework-MobSF/), while assessing an application. An application can be assessed by side-loading it, re-packaging it, or by simply attacking the installed version.
 
 ##### Using Non-Rooted Devices
 
-##### Method Tracing
+Non-rooted devices provide the tester with two benefits:
+
+- Replicate an environment that the application is intended to run on
+- As a tester you don't need to bypass root detection and can use objection to package the app with the Frida Gadget.
```

Comment:
Split into 2 benefits

```suggestion
- If root detection is in place you don't need to bypass it because of running on a legitimate environment.
- Thanks to tools like objection you can patch the app to able able to test like if you were on a rooted device (but of course being jailed to that one app).
```

Reply:
Hmm reading it again now, don't you think that line 1260 is redundant? If that's the environment that it is intended to run on, then it won't need root bypass.

Reply:
Yes, maybe you can leave that line out and leave the other two ;)